src/bootstrap/lib.rs

```diff
@@ -65,9 +65,15 @@
 //! also check out the `src/bootstrap/README.md` file for more information.
 
 #![deny(warnings)]
+#![feature(associated_consts)]
+#![feature(core_intrinsics)]
```

Comment:
> Can't this be done with trait dispatch?

Not sure how you'd integrate that with needing the function to be generic, given that's forbidden in traits? Maybe you have something else in mind.